### PR TITLE
(fix) O3-4253: Add missing translations support in the app

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR adds the missing translation implementation in the following places:
* Translations should be extracted from `index.ts`
* The dashboard link didn't support the translation of the dashboard title
* Renamed `date-range-picker.tsx` to `date-range-picker.component.tsx` to extract translations from the file
* Add translation keys for the lab order panel tabs

### Before
image

### After
image

## Related Issue
https://issues.openmrs.org/browse/O3-4253

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -36,32 +37,37 @@ export const rejectLabRequestModal = getAsyncLifecycle(
...
   options,
 );
 
+// t('Worklist', 'Worklist')
 export const worklistTile = getAsyncLifecycle(
   () => import('./lab-tiles/in-progress-lab-requests-tile.component'),
   options,
 );
 
+// t("Referred Tests", "Referred Tests")
```
Comment (denniskigen) [inline src/index.ts:64]:
```suggestion
// t("Referred Tests", "Referred tests")
```

**Review by denniskigen — COMMENTED** [review]
Looks good. Just one suggested change for the `Referred tests` translation.

**Comment by vasharma05** [comment]
Hi @denniskigen, the change is made, please re-review the PR.
Thanks!

**Review by denniskigen — APPROVED** [review]
Thanks!

**Pull request merged**